paasta_tools/cli/cmds/spark_run.py

```diff
@@ -455,10 +478,8 @@ def get_docker_cmd(args, instance_config, spark_conf_str):
 
     # Default cli options to start the jupyter notebook server.
     if original_docker_cmd == 'jupyter':
-        # Shutdown idle kernels that are not connected after one hour.
-        # Shutdown connected kernels if they use more than 32 cores.
-        cull_opts = '--MappingKernelManager.cull_idle_timeout=3600 '
-        if args.max_cores > 32:
+        cull_opts = '--MappingKernelManager.cull_idle_timeout=%s ' % args.cull_idle_timeout
+        if args.cull_connected_max_cores > 0 and args.max_cores > args.cull_connected_max_cores:
```

Comment:
It feels a little weird to me that cull_connected_max_cores is configurable by the user -- in what circumstance would we expect to see someone run with, say, `--max-cores=100 --cull-connected-max-cores=50`?  Maybe we should just expose a `--cull-connected` option, and if it's not specified we only turn it on if `max-cores` is over 32?

Reply:
That was my first impression too. Potentially useful if the user is using some default configuration on top of just this CLI. For example an alias or script which accepts a variable `max_cores` but wants a higher default for this cull option.

Related, would an option accepting a profile from a yaml/json config as opposed to recording working cli argument combinations for a given job be useful?